src/pages/storage/StorageDetailHeader.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { FC, useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import RenameHeader, { RenameHeaderValues } from "components/RenameHeader";
+import { useNotify } from "context/notify";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import { LxdStoragePool } from "types/storage";
+import { renameStoragePool } from "api/storage-pools";
+import DeleteStorageBtn from "pages/storage/actions/DeleteStorageBtn";
+import { testDuplicateName } from "util/storage";
+
+interface Props {
+  name: string;
+  storagePool: LxdStoragePool;
+  project: string;
+}
+
+const StorageDetailHeader: FC<Props> = ({ name, storagePool, project }) => {
+  const navigate = useNavigate();
+  const notify = useNotify();
+  const controllerState = useState<AbortController | null>(null);
+
+  const RenameSchema = Yup.object().shape({
+    name: Yup.string()
+      .test(...testDuplicateName(project, controllerState))
+      .required("This field is required"),
+  });
+
+  const formik = useFormik<RenameHeaderValues>({
+    initialValues: {
+      name,
+      isRenaming: false,
+    },
+    validationSchema: RenameSchema,
+    onSubmit: (values) => {
+      if (name === values.name) {
+        void formik.setFieldValue("isRenaming", false);
+        formik.setSubmitting(false);
+        return;
+      }
+      renameStoragePool(name, values.name, project)
+        .then(() => {
+          navigate(
+            `/ui/${project}/storage/${values.name}`,
+            notify.queue(notify.success("Storage pool renamed."))
+          );
+          void formik.setFieldValue("isRenaming", false);
+        })
+        .catch((e) => {
+          notify.failure("Renaming failed", e);
+        })
+        .finally(() => formik.setSubmitting(false));
+    },
+  });
+
+  return (
+    <RenameHeader
+      name={name}
+      parentItem={<Link to={`/ui/${project}/storage`}>Storage pools</Link>}
+      controls={
+        <DeleteStorageBtn
+          key="delete"
+          storage={storagePool}
+          project={project}
+        />
+      }
+      isLoaded={true}
+      formik={formik}
+      renameDisabledReason="Cannot rename storage pools"
```

Comment:
This disables the rename of storage pools completely - is there a reason why we want to do that? And if we do want to disable it, why do we need the API function, the formik etc.? It's confusing because the infrastructure makes you think that you can rename, but in fact you can't.

Reply:
The API gave me a "feature not implemented" error on renaming of storage pools. I assume it is a thing that is planned for later. Hence I added support for it with this message.